### PR TITLE
Kernel+Userland: Add support for using the PCSpeaker with various tones

### PR DESCRIPTION
**Pull request by supercomputer7**

### CODE REVIEW

File: Userland/Utilities/beep.cpp [hunk 1]
```diff
@@ -4,11 +4,16 @@
...
 #include <LibCore/System.h>
 #include <LibMain/Main.h>
 
-ErrorOr<int> serenity_main(Main::Arguments)
+ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    TRY(Core::System::beep());
+    Optional<size_t> tone;
+    Core::ArgsParser args_parser;
+    args_parser.add_option(tone, "Beep tone", "beep-tone", 'n', "Beep tone (frequency in Hz)");
```
Comment (AtkinsSJ) [inline Userland/Utilities/beep.cpp:15]: The `beep` utility on Linux uses `-f` for setting the frequency. Seems like BSD does too? So IMO it'd be better to use that for consistency.

Comment (supercomputer7) [reply Userland/Utilities/beep.cpp]: Done :)

File: Userland/Utilities/beep.cpp [hunk 2]
```diff
@@ -4,11 +4,16 @@
...
 
-ErrorOr<int> serenity_main(Main::Arguments)
+ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    TRY(Core::System::beep());
+    Optional<size_t> tone;
+    Core::ArgsParser args_parser;
+    args_parser.add_option(tone, "Beep tone", "beep-tone", 'f', "Beep tone (frequency in Hz)");
+    args_parser.parse(arguments);
+    TRY(Core::System::beep(tone.value_or(440)));
```
Comment (konradekk) [inline Userland/Utilities/beep.cpp:17]: it looks sensible for me to have a default value in the syscall itself… 🤔 (i.e. i want a beep from the system; not knowing exactly which is the best—or even that there multiple possible—to think of it: the time should be also set! 😄)

Comment (AtkinsSJ) [reply Userland/Utilities/beep.cpp:17]: Yeah +1 for making `Core::System::beep()` default to 440.

**Comment by hjalves** [comment]
Very cool! Just tried this on bare metal.

File: Userland/Utilities/beep.cpp (hunk 2 above)
Comment (konradekk) [reply Userland/Utilities/beep.cpp]: well… i had a `sys$beep` itself in mind – not only the core `beep`… 😟

**Pull request merged**